solver/llbsolver/file/backend.go

```diff
@@ -100,6 +100,27 @@ func mkfile(ctx context.Context, d string, action pb.FileActionMkFile, user *cop
 }
 
 func rm(ctx context.Context, d string, action pb.FileActionRm) error {
+	if action.AllowWildcard {
+		src := cleanPath(action.Path)
+		m, err := copy.ResolveWildcards(d, src, false)
+		if err != nil {
+			return err
+		}
+
+		for _, s := range m {
+			p, err := fs.RootPath(d, filepath.Join(filepath.Join("/", s)))
```

Comment:
Maybe create an extra function for this to avoid repetition with non-wildcard path.